Project-II-Calculator/calculator/src/App.js

```diff
@@ -1,23 +1,138 @@
-      </p>
-      <p>
-        <strong>
-          Don't forget to `default export` your components and import them here
-          inside of this file in order to make them work.
-        </strong>
-      </p>
+    <div className = "calculator">
+        <CalculatorDisplay content = {total}/>
+        <div className = "button-groups">
+            <div className = "left-button-group">
+              <ActionButton content = "clear" style ="action-button-top" function = {() =>updateTotal("clear")}/>
+              <NumberPanel />
+              <ActionButton content = "0" style = "action-button-bottom" function = {() =>updateTotal("0")}/>
+            </div>
+
+          <div className = "right-button-group">
+              <NumberButton buttonStyle = "function-button" content = {"\u00f7"} function = {()=>updateTotal("\u00f7")}/>
+              <NumberButton buttonStyle = "function-button" content = {"\u00d7"} function = {()=>updateTotal("\u00d7")}/>
+              <NumberButton buttonStyle = "function-button" content = {"-"} function = {()=>updateTotal("-")} />
+              <NumberButton buttonStyle = "function-button" content = {"+"} function = {()=>updateTotal("+")} />
+              <NumberButton buttonStyle = "function-button" content = {"="} function = {()=>updateTotal("=")} />
+
+
+
+
+
+          </div>
+        </div>
     </div>
   );
 };
+*/
+
+class App extends React.Component{
+  constructor(){
+    super();
+    this.state = {
+      total: 0,
+      display: 0
+    };
+  }
+
+   updateTotal = (input) =>{
+    console.log(input);
+      if(input==="="){
+        total = eval(formula);
+        console.log(total);
+        this.setState({total: total});
+        this.setState({display:total});
+
+        formula = "";
+        displayFormula="";
+        return;
+
+      }
+      if(input ==="\u00d7"){
+        formula+="*";
+        displayFormula+=input;
+        this.setState({display:displayFormula});
+
+        return;
+      }
+      if(input ==="\u00f7"){
+        formula+="/";
+        displayFormula+=input;
+        this.setState({display:displayFormula});
+
+        return;
+      }
+      if(input ==="clear"){
+        this.setState({total: 0});
+        this.setState({display: 0});
+        formula = "";
+        displayFormula="";
+
+        return;
+
+      }
+      else{
+        formula+=input;
+        displayFormula +=input;
+        this.setState({display:displayFormula});
+        return;
+
+      }
+  }
+
+  listNums(){
+    const numButtons= nums.map(num => {
+      return <NumberButton buttonStyle = "number-button" content= {num} function = {()=>this.updateTotal(num)} />;
+
+    });
+    return <div className = "num-buttons-group">{numButtons}</div>;
+
+  }
+
+
+
+
+  render(){
+    return(  <div className = "calculator">
+          <CalculatorDisplay content = {this.state.display}/>
+          <div className = "button-groups">
+              <div className = "left-button-group">
```

Comment:
How could you have made this more DRY?
Could you have passed your methods through and used a functional component?